Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -35,39 +37,284 @@ struct DeclarationsSectionTranslator: RenderSectionTranslator {
+                        // break whitespace into their own blocks
+                        return false
+                    } else if currentElement == "?" {
+                        // if we have a token like `?>` or similar we should break the fragment
+                        return false
+                    } else {
+                        return true
+                    }
+                }
+
+                // FIXME: replace this with `chunked(by:)` if we add swift-algorithms as a dependency
+                for (nextIndex, nextElement) in fragment.spelling.indexed().dropFirst() {
+                    if !areInSameChunk(currentElement, nextElement) {
+                        textPartitions.append(String(fragment.spelling[substringIndex..<nextIndex]))
+                        substringIndex = nextIndex
+                    }
+                    currentElement = nextElement
+                }
+
+                if substringIndex != fragment.spelling.endIndex {
+                    textPartitions.append(String(fragment.spelling[substringIndex...]))
+                }
+
+                return textPartitions.map({ .init(kind: .text, spelling: $0, preciseIdentifier: nil) })
             }
 
             var declarations = [DeclarationRenderSection]()
             for pair in declaration {
                 let (platforms, declaration) = pair
 
-                let renderedTokens = declaration.declarationFragments.map(translateFragment)
+                let renderedTokens: [DeclarationRenderSection.Token]
+                let otherDeclarations: DeclarationRenderSection.OtherDeclarations?
+
+                // If this symbol has overloads, render their declarations as well.
+                if let overloads = symbol.overloadsVariants[trait],
+                   let overloadDeclarations = symbol.overloadsVariants[trait]?.references.compactMap({ overloadReference -> OverloadDeclaration? in
+                    guard let overload = try? renderNodeTranslator.context.entity(with: overloadReference).semantic as? Symbol else {
+                        return nil
+                    }
+
+                    let declarationFragments = overload.declarationVariants[trait]?.values.first?.declarationFragments
+                    assert(declarationFragments != nil, "Overloaded symbols must have declaration fragments.")
+                    return declarationFragments.map({ (declaration: $0, reference: overloadReference )})
+                }), !overloadDeclarations.isEmpty {
+                    // Pre-process the declarations by splitting text fragments apart to increase legibility
+                    let mainDeclaration = declaration.declarationFragments.flatMap(preProcessFragment(_:))
+                    let processedOverloadDeclarations = overloadDeclarations.map({ ($0.declaration.flatMap(preProcessFragment(_:)), $0.reference) })
+                    let preProcessedDeclarations = [mainDeclaration] + processedOverloadDeclarations.map(\.0)
```

Comment:
I think the distinction is minimal. Even SymbolKit doesn't really distinguish between the concept of a "declaration" and "a list of declaration fragments"; the convenience `Symbol.declarationFragments` accessor directly returns the list instead of the wrapper object. It also avoids having to pluralize a plural, since `processedOverloadDeclarations` and `preProcessedDeclarations` are lists of lists. To make them reference "fragments" instead, it would have to be something like `processedOverloadFragmentLists` or the like, which i consider more awkward.